runtime-modules/content/src/migrations.rs

```diff
@@ -0,0 +1,219 @@
+
+        // Group's moderation permissions (by channel's privilage level)
+        pub permissions_by_level: ModerationPermissionsByLevel,
+    }
+
+    type CuratorGroupV0<T> =
+        CuratorGroupRecordV0<CuratorGroupCuratorsMap<T>, StoredModerationPermissionsByLevelV0<T>>;
+
+    type StoredCuratorModerationPermissionsV0 =
+        BoundedBTreeSet<iterable_enums::ContentModerationActionV0, MaxCuratorPermissionsPerLevelV0>;
+
+    type StoredModerationPermissionsByLevelV0<T> = BoundedBTreeMap<
+        <T as Config>::ChannelPrivilegeLevel,
+        StoredCuratorModerationPermissionsV0,
+        <T as Config>::MaxKeysPerCuratorGroupPermissionsByLevelMap,
+    >;
+
+    // #[frame_support::storage_alias]
+    // type CuratorGroupByIdV0<T: Config> = StorageMap<
+    //     Pallet<T>,
+    //     Blake2_128Concat,
+    //     <T as permissions::ContentActorAuthenticator>::CuratorGroupId,
+    //     CuratorGroupV0<T>,
+    // >;
+
+    pub struct MigrateToV1<T>(sp_std::marker::PhantomData<T>);
+    impl<T: Config> OnRuntimeUpgrade for MigrateToV1<T> {
+        #[cfg(feature = "try-runtime")]
+        fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+            let onchain = Pallet::<T>::on_chain_storage_version();
+
+            ensure!(onchain < 1, "this migration can be deleted");
+
+            // using storage alias CuratorGroupByIdV0 was not working.
+            // most likely wrong prefix is getting used, since even iterating over keys return 0 count.
+            // So we use the new map but only iterate over keys to avoid decode errors. (if decode fails the item is skipped)
+            let group_count: usize = CuratorGroupById::<T>::iter_keys().count();
+            log!(info, "Number of curator groups pre_upgrade {}", group_count);
+
+            Ok(group_count.to_be_bytes().to_vec())
+        }
+
+        fn on_runtime_upgrade() -> Weight {
+            let onchain = Pallet::<T>::on_chain_storage_version();
+            let current = Pallet::<T>::current_storage_version();
+
+            if onchain > 0 {
+                return T::DbWeight::get().reads(1);
+            }
+
+            use iterable_enums::ContentModerationActionV0;
+
+            <CuratorGroupById<T>>::translate_values(
+                |old_value: CuratorGroupV0<T>| -> Option<CuratorGroup<T>> {
+                    log!(info, "Migrating group {:?}", old_value);
+                    let mut permissions_by_level = ModerationPermissionsByLevel::<T>::new();
+                    old_value
+                        .permissions_by_level
+                        .iter()
+                        .for_each(|(level, permissions)| {
+                            log!(info, "level: {:?}, permissions: {:?}", level, permissions);
+                            let mut permissions_v1 = BTreeSet::new();
+                            permissions.iter().for_each(|action| match action {
+                                ContentModerationActionV0::HideVideo => {
```

Comment:
I think it would have been better to separate out conversion from writing/translation, but don't think its worth fiddling with now.